From 555a3804c8f46ffab4e3bfa3253948065ac6f13b Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Fri, 11 Jan 2019 12:21:05 +0900
Subject: [PATCH 745/909] drm: rcar-du: Fix loop procedure in scatter gather
 page set

commit 5a5e59b87990d7f3d91a9e19ccaecbec0896cd1f from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Originally the variable of "i" is used for the loop count of
the number of format planes in rcar_du_vsp_plane_prepare_fb
function and should be not used the same "i" as the loop count
of sg_set_page.

It is affected when the number of format planes is 2 (NV12 format
etc.) and 3 (YUV420 format etc.).
For the number of format planes is 1 (RGB565, ARGB888, YUYV etc),
this function works fine.

This patch solves its problem.

Fixes: c424cd4724 ("drm: rcar-du: Allow importing non-contiguous
dma-buf with VSP")

Reviewed-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_vsp.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_vsp.c b/drivers/gpu/drm/rcar-du/rcar_du_vsp.c
index 4c56f4e961ca..ea35f8c86e88 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_vsp.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_vsp.c
@@ -1,7 +1,7 @@
 /*
- * rcar_du_vsp.h  --  R-Car Display Unit VSP-Based Compositor
+ * rcar_du_vsp.c  --  R-Car Display Unit VSP-Based Compositor
  *
- * Copyright (C) 2015-2018 Renesas Electronics Corporation
+ * Copyright (C) 2015-2019 Renesas Electronics Corporation
  *
  * Contact: Laurent Pinchart (laurent.pinchart@ideasonboard.com)
  *
@@ -227,7 +227,7 @@ static int rcar_du_vsp_plane_prepare_fb(struct drm_plane *plane,
 	struct rcar_du_vsp_plane_state *rstate = to_rcar_vsp_plane_state(state);
 	struct rcar_du_vsp *vsp = to_rcar_vsp_plane(plane)->vsp;
 	struct rcar_du_device *rcdu = vsp->dev;
-	unsigned int i;
+	unsigned int i, j;
 	int ret;
 
 	/*
@@ -260,7 +260,7 @@ static int rcar_du_vsp_plane_prepare_fb(struct drm_plane *plane,
 
 			src = gem->sgt->sgl;
 			dst = sgt->sgl;
-			for (i = 0; i < gem->sgt->orig_nents; ++i) {
+			for (j = 0; j < gem->sgt->orig_nents; ++j) {
 				sg_set_page(dst, sg_page(src), src->length,
 					    src->offset);
 				src = sg_next(src);
-- 
2.17.1

